From 5277ae626bd2edf03a95aa4a58982c8c2493a1f0 Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Thu, 2 Oct 2025 16:17:05 +0200
Subject: [PATCH 403/553] Mark qcbor{map,array}.cpp as security-critical

QJsonArray/Map are, too, because the whole resp. component is, and the
implementation in those two files cannot be said to be trivial.

Amends cd4813a3c304a6d9d5d8096d91a8fc297818d624.

QUIP: 23
Task-number: QTBUG-135194
Pick-to: 6.8
Change-Id: I08dfc1dc1f917832f0cd24d3b575095c5e930a4d
Reviewed-by: Edward Welbourne <edward.welbourne@qt.io>
(cherry picked from commit 9b5313a98f4a971919ead662a113f5adb20ef766)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/serialization/qcborarray.cpp | 1 +
 src/corelib/serialization/qcbormap.cpp   | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/corelib/serialization/qcborarray.cpp b/src/corelib/serialization/qcborarray.cpp
index 9a566b999b4..de6edcb75e6 100644
--- a/src/corelib/serialization/qcborarray.cpp
+++ b/src/corelib/serialization/qcborarray.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2018 Intel Corporation.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:critical reason:data-parser
 
 #include "qcborarray.h"
 #include "qcborvalue_p.h"
diff --git a/src/corelib/serialization/qcbormap.cpp b/src/corelib/serialization/qcbormap.cpp
index 32d6d563ac4..9c17e07b70b 100644
--- a/src/corelib/serialization/qcbormap.cpp
+++ b/src/corelib/serialization/qcbormap.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2018 Intel Corporation.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:critical reason:data-parser
 
 #include "qcbormap.h"
 #include "qcborvalue_p.h"
-- 
2.51.2

